Dockerfile

```diff
@@ -22,8 +22,12 @@ FROM amazoncorretto:21
 
```

Comment:
I wonder if we can use Ubuntu or CentOS here.  
It seems that we can't specify the PostgreSQL version to install with Amazon Corretto. I might need to install PostgreSQL client version 9.6, which matches our PostgreSQL server (as specified in Docker Compose). Otherwise I would get encounter a `pg_dump` version mismatch error when trying to get an SQL dump from PostgreSQL.

Reply:
I don't think the postgresql client version has to match with the postgresql server version. Otherwise we have an issue anyways since the postgresdump should work with any postgres version (or at least all the recent ones).

The postgres version in the docker-compose is just a random one anyways, I should probably use a newer one.

I am using amazoncorretto because that is a openjdk base docker image maintained by one of the bigger players, and unlikely to go anywhere soon. If we start with ubuntu, we'd have to manually install everything for the JVM etc.

Reply:
I didn’t think the version had to match, similar to how `mysqldump` usually works with any MySQL version. However, it seems that’s not the case with `pg_dump`. Even though I updated the PostgreSQL server to version 16, I still got the error. Also I think Amazon Corretto offers PostgreSQL 9.6 by default.

![image](https://github.com/user-attachments/assets/286cd492-8fb0-4def-bbd6-b4a4e0f0c291)

Reply:
> Because pg_dump is used to transfer data to newer versions of PostgreSQL, the output of pg_dump can be expected to load into PostgreSQL server versions newer than pg_dump's version. pg_dump can also dump from PostgreSQL servers older than its own version. (Currently, servers back to version 8.0 are supported.) However, pg_dump cannot dump from PostgreSQL servers newer than its own major version; it will refuse to even try, rather than risk making an invalid dump. Also, it is not guaranteed that pg_dump's output can be loaded into a server of an older major version — not even if the dump was taken from a server of that version. Loading a dump file into an older server may require manual editing of the dump file to remove syntax not understood by the older server. Use of the --quote-all-identifiers option is recommended in cross-version cases, as it can prevent problems arising from varying reserved-word lists in different PostgreSQL versions.


From the docs: https://www.postgresql.org/docs/14/app-pgdump.html
This is not ideal, but I think we just install the newest possible version and use the `--quote-all-identifiers` flag and then hope for the best.

This is a good indication thought that for a more robust setup an extra dumping container might be a better idea. Then we could e.g. load a postgres:`some.version` container and run the exact pg_dump version for that server version, without holding all the dependencies in the Kviklet container directly.

Reply:
Hi Jascha, sorry for my late reply; I was on vacation.

Did you mean to install the latest version of `pg_dump`? I tried this previously and found that it seems we can't specify the PostgreSQL version to install with Amazon Corretto.

Regarding adding a container for the dump, I tried the following:
In the backend `ExecutionRequestService.kt` file, in the `constructSQLDumpCommand` method, I changed the `sqlDumpCommand` from:

```
DatasourceType.POSTGRESQL -> {
    val sqlDumpCommand = listOfNotNull(
        "pg_dump",
        "-U${connection.username}",
        "-h${connection.hostname}",
        "-p${connection.port}",
        connection.databaseName,
        outputFile?.let { "--file=$it" },
    )
    sqlDumpCommand
}
```
to:

```
DatasourceType.POSTGRESQL -> {
    val sqlDumpCommand = listOfNotNull(
        "docker", "exec", "pg_dump_service", "pg_dump",  // Updated for PostgreSQL
        "-U${connection.username}",
        "-h${connection.hostname}",
        "-p${connection.port}",
        connection.databaseName,
        outputFile?.let { "--file=$it" },
    )
    sqlDumpCommand
}

```
I also added the `pg_dump_service` container to `docker-compose`:

```
pg_dump_service:
    image: postgres:16.3
    environment:
      POSTGRES_PASSWORD: postgres
    volumes:
      - ./dumps:/dumps
    depends_on:
      - kviklet-postgres

kviklet-postgres:
    image: postgres:9.6
    restart: always
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: kviklet
    ports:
      - "5433:5432"
    volumes:
      - ./kviklet-postgres-data:/var/lib/postgresql/data

kviklet:
    # image: ghcr.io/kviklet/kviklet:main
    build: .
    ports:
      - "80:80"
    environment:
      - SPRING_DATASOURCE_URL=jdbc:postgresql://kviklet-postgres:5432/kviklet
      - SPRING_DATASOURCE_USERNAME=postgres
      - SPRING_DATASOURCE_PASSWORD=postgres
      - INITIAL_USER_EMAIL=admin@admin.com
      - INITIAL_USER_PASSWORD=admin
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - ./dumps:/dumps
    depends_on:
      - kviklet-postgres
```

Docker installation is also included in the Dockerfile. However, it did not work, and I believe running Docker inside Docker should be avoided.

I'm not sure if the approach above is what you had in mind or if you were thinking of another solution?

Overall, I found that having the dumping feature for PostgreSQL is more challenging than expected and might require additional time to resolve. I was wondering if it would be acceptable to enable the dump feature for just MySQL for now?

Reply:
Yeh no, I don't want to run docker in docker for sure. And yes sure, I'm fine with doing MySQL only for now. I'll have a look at a later point in time on how to do the psql dump.

Reply:
Nice! I'll let you know if I find something that could potentially work. Meanwhile, do you have any other feedback on this PR? Let me know if there's anything else that needs to be addressed. Thank you!